public/app/plugins/datasource/grafana-azure-monitor-datasource/variables.test.ts

```diff
@@ -36,43 +37,11 @@ describe('VariableSupport', () => {
           } as AzureMonitorQuery,
         ],
       } as DataQueryRequest<AzureMonitorQuery>;
-      const observables = variableSupport.query(mockRequest);
-      observables.subscribe((result: DataQueryResponseData) => {
-        expect(result.data[0].source).toEqual(fakeSubscriptions);
-        done();
-      });
-    });
-
-    it('can fetch resourceGroups with a default subscriptionId', (done) => {
-      const expectedResults = ['test'];
-      const variableSupport = new VariableSupport(
-        createMockDatasource({
-          azureLogAnalyticsDatasource: {
-            defaultSubscriptionId: 'defaultSubscriptionId',
-          },
-          getResourceGroups: jest.fn().mockResolvedValueOnce(expectedResults),
-        })
-      );
-      const mockRequest = {
-        targets: [
-          {
-            refId: 'A',
-            queryType: AzureQueryType.GrafanaTemplateVariableFn,
-            grafanaTemplateVariableFn: {
-              kind: 'ResourceGroupsQuery',
-              rawQuery: 'ResourceGroups()',
-            },
-          } as AzureMonitorQuery,
-        ],
-      } as DataQueryRequest<AzureMonitorQuery>;
```

Comment:
these tests `can ... with a default subscription` were returning false positives. According to functional code, requests with a `grafanaTemplateVariableFn` object (which is deprecated), should already have all the required parameters in that object. This `mockRequest` is an invalid request, `subscription` (and other parameters) should already be there.

The test was passing because it was taking advantage of a bug (I'll explain the bug in other comment).

It's the process of migrating a query from a string (even more legacy) to an object the one that gets the default subscription and adds it to the `grafanaTemplateVariableFn` object.

Reply:
I love the removal of invalid tests, great stuff here 😊